### ISSUE TITLE
Replace PyQwt with PyQtGraph

### ISSUE THREAD

**Issue by noc0lour** [gnuradio/gnuradio#1634]
PyQwt is effectively unmaintained since 2014 and thus has no support for Qt5/Qwt6 and Python3. 
Fortunately there is PyQtGraph which could be used to replace all occurences of PyQwt.

Occurences:  
 - [ ] gr-filter/python/filter/design/filter_design.py
 - [ ] gr-filter/python/filter/gui/idealbanditems.py
 - [ ] gr-filter/python/filter/gui/pyqt_filter_stacked.py
 - [ ] gr-filter/python/filter/gui/polezero_plot.py
 - [ ] gr-qtgui/apps/plot_form.py
 - [ ] gr-qtgui/apps/plot_time_form.py
 - [ ] gr-utils/python/utils/pyqt_filter.py
 - [ ] gr-utils/python/utils/pyqt_plot.py
 - [ ] gr-zeromq/examples/python/gui.py

**Comment by ekigwana** [comment]
Any progress on this? It'd be nice to be able to use the filter design tool.

**Comment by noc0lour** [comment]
I barely looked into it, any help is appreciated.

**Comment by ekigwana** [comment]
I have started to look at this and having only PyQt5 I have noticed that a few other changes will be required. From PyQt5  [gotchas](http://pyqt.sourceforge.net/Docs/PyQt5/gotchas.html) I noticed QStrings are gone and PyQt5 is not compatible with PyQt5. I forked the next branch and I believe it is targeting PyQT5 and Python v2, is that correct? Is PyQt4 support also required?

**Comment by noc0lour** [comment]
If possible I'd recommend to have development in py2/py3 compat mode. With GNU Radio 3.8 the goal is to support py3 as well.

**Comment by ekigwana** [comment]
To keep things reasonable since QT4 is EOL and I don't even have a system with it, I am targeting the following:

1. Python v2/v3
2. PyQt5
3. pyqtgraphics

**Comment by ekigwana** [comment]
I have been working on the filter design tool and have a question below.

New "Filter Taps" plot in progress unless it is not required
![filter taps_new](https://user-images.githubusercontent.com/13283456/41314665-fb023912-6e7c-11e8-9461-aa9855da7b77.png)
Old "Filter Taps" plot
![filter taps_old](https://user-images.githubusercontent.com/13283456/41314683-0637b4ce-6e7d-11e8-9405-ef00b0565d98.png)
New "Impulse Response" plot
![impulse_response_new](https://user-images.githubusercontent.com/13283456/41314693-0b34bc56-6e7d-11e8-81ab-1193fff7d90c.png)
Old "Impulse Response" plot
![impulse_response_old](https://user-images.githubusercontent.com/13283456/41314694-0b51c008-6e7d-11e8-9041-83149ed5b7f3.png)
New "Step Response" plot
![step_response_new](https://user-images.githubusercontent.com/13283456/41314695-0b72456c-6e7d-11e8-9245-9d55afbd53cc.png)
Old "Step Response" plot
![step_response_old](https://user-images.githubusercontent.com/13283456/41314696-0bad40e0-6e7d-11e8-9724-8f2db8597af9.png)

What is the purpose of the "Filter Taps" plot? It is exactly the same as the "Impulse Response" but with a different x axis label and different tab label. It looks pretty useless to me as an Electrical engineer and think it should go. I am going some space between the view border and the axis labels so not done yet.

**Comment by marcusmueller** [comment]
@ekigwana somehow this slipped my attention. Is your code somewhere that I can look at it and merge it?

**Comment by noc0lour** [comment]
I tried the `pyqtgraph` branch on the @ekigwana remote, I actually rebased that on the new master. Maybe that helps with springstarting this development again?

**Comment by ekigwana** [comment]
Sorry I got too busy. From what I could tell it should be "fairly" easy.

**Comment by btashton** [comment]
@noc0lour @ekigwana Anything I can do to help here?

**Comment by noc0lour** [comment]
@btashton Sure, if you wish you can pick my branch and open a separate PR for tracking/discussing your progress on that matter. Other than @marcusmueller rejecting `pyqtgraph` as dependency I'd say go for it.

**Comment by btashton** [comment]
@marcusmueller mentioned here #2145 that pyqtgraph might be up for discussion.

**Comment by btashton** [comment]
@marcusmueller @noc0lour before I look into picking this up too much, are we going to have a CLA issue, I don't want to spend too much time and then get blocked by needing a CLA from @ekigwana

**Comment by noc0lour** [comment]
Good question. My gut feeling tells me we might need one from @ekigwana . @bhilburn ?

**Comment by ekigwana** [comment]
I really just ran out of time to work on this but I'll give you whatever you need

**Comment by ekigwana** [comment]
Also when I looked at the filter design code, it looked like someone was transitioning to a modern interface so there is otherwise quite a bit of dead code sitting in there. I like the look of https://github.com/chipmuenk/pyFDA but it uses matplotlib although the concept still works a bit better for that. For this to work, the issues raised in #2145 would certainly need to be addressed in a project wide manner. My two cents FWIW.

**Comment by bhilburn** [comment]
Yes, we'll need a CLA from @ekigwana. Can you e-mail me? bhilburn@gnuradio.org

**Comment by ekigwana** [comment]
edwardwwgk@gmail.com

**Comment by btashton** [comment]
Sweet, I'll pick the rest of the filter stuff up once this is done.  The plotting stuff I think is in a good place, just need to move the code over and add the file seek interface to the GUI.

**Comment by bhilburn** [comment]
@ekigwana - Just sent you an e-mail with instructions for getting the CLA in place!
